securedrop/crypto_util.py

```diff
@@ -77,8 +77,38 @@ def clean(s, also=''):
     return str(s)
 
 
-def genrandomid(words_in_random_id=DEFAULT_WORDS_IN_RANDOM_ID):
-    return ' '.join(random.choice(words) for x in range(words_in_random_id))
+def _get_wordlist(locale):
+    """" Ensure the wordlist for the desired locale is read and available
+    in the words global variable. If there is no wordlist for the
+    desired local, fallback to the default english wordlist.
+
+    The localized wordlist are read from wordlists/{locale}.txt but
+    for backward compatibility purposes the english wordlist is read
+    from the config.WORD_LIST file.
+
+    """
+
+    if locale not in language2words:
+        if locale != 'en':
+            path = os.path.join(config.SECUREDROP_ROOT,
+                                'wordlists',
+                                locale + '.txt')
+            if os.path.exists(path):
+                wordlist_path = path
+            else:
+                wordlist_path = config.WORD_LIST
+        else:
+            wordlist_path = config.WORD_LIST
+
+        language2words[locale] = open(
+            wordlist_path).read().rstrip('\n').split('\n')
+
+    return language2words[locale]
+
+
+def genrandomid(words_in_random_id=DEFAULT_WORDS_IN_RANDOM_ID, locale='en'):
+    return ' '.join(random.choice(_get_wordlist(locale))
+                    for x in range(words_in_random_id))
```

Comment:
On the journalist side the source's `journalist_designation` will continue to be two English words - because the two wordlists used for this designation are in English and are separate from the wordlist here (this doesn't block merge, just noting).

Reply:
Yes, it could be improved in a separate pull request. The focus of this pull request is to have passphrases that are easier to remember when a localized wordlist is available. And since the journalist does not need to remember the source designation, it is less of a problem if it always is in english.